lib/PipelineC/PipelineC.cpp

```diff
@@ -807,7 +807,9 @@ const char *rp_container_get_mime(rp_container *container) {
 
 const char *
 rp_container_extract_one(rp_container *container, rp_target *target) {
-  revng_check(container->second->enumerate().contains(*target));
+  if (!container->second->enumerate().contains(*target)) {
+    return nullptr;
+  }
```

Comment:
> Fix PipelineC aborting when it should return nullptr

The docs says otherwise:

```
include/revng/PipelineC/Prototypes.h-/**
include/revng/PipelineC/Prototypes.h- * \return the serialized content of the element associated to the provided
include/revng/PipelineC/Prototypes.h- * target,
include/revng/PipelineC/Prototypes.h- *
include/revng/PipelineC/Prototypes.h- * \note Target must be already present in container
include/revng/PipelineC/Prototypes.h- */
include/revng/PipelineC/Prototypes.h-const char * /*owning*/
include/revng/PipelineC/Prototypes.h:rp_container_extract_one(rp_container *container, rp_target *target);
```

Do we have an API to check if the target is available before calling this?
If you really want to return `nullptr`, we need to update the doc.

I know you don't like `abort`s, but remember that this is an internal API, it's not going to be used by users, so forcing ourselves to do extra checks is not too bad.

Reply:
There's `rp_target_is_ready` but I'd argue that it's more consistent with the rest of the API where `nullptr` says it's not present

Reply:
OK, then update the doc.

Reply:
Updated